### ISSUE TITLE
gf_mul duplicate symbol clash with libcrypto

### ISSUE THREAD

**Issue by anders-wind** [intel/isa-l#217]
Hi we are trying to use both ISA-L and aws-sdk (and apache Arrow) both of which includes libcrypto. 
We are getting a duplicate symbol error on `gf_mul`. Would it be possible to rename `gf_mul`?. The issue seems to only occur with gcc when building with `--coverage` which does not prune all unused symbols.

We are using vcpkg for loading in both isal and the aws-sdk.

**Comment by gbtucker** [comment]
That would be difficult as it is an external function for ISA-L and part of the api.  It looks like for openssl, it is an internal function and not part of their external exposed api.

I made a test by combining our isa-l and isa-l_crypto openssl comparisons and I can link it with both libisal and libcrypto without issue.  I can even compile the test with `--coverage` and the isa-l library with `--coverage` and see no build, link or runtime warnings.  Are you building the entire openssl library with --coverage on?

**Comment by anders-wind** [comment]
Ill come up with a reproducible example :) 
I'm using vcpkg for both libcrypto and ISA-l, so I don't think the compile flag is being added to either of the dependencies but the linker flag might affect them.
```
    "CMAKE_CXX_FLAGS_DEBUG": "-g -Og --coverage",
    "CMAKE_EXE_LINKER_FLAGS": "--coverage"
 ```

In any case even though I think `gf_mul` seems like a very common name to put in a library's api, I understand that, thats just how it is. Ill try to contact openssl :)

**Comment by gbtucker** [comment]
Agreed that gf_mul was not the best name.  I can't reproduce with the above flags either though.

**Comment by anders-wind** [comment]
I could reproduce when also using `add_link_options("-fuse-ld=lld")` on top of `--coverage` btw. Not saying you should do anything about it however :)
